pass/provider.go

```diff
@@ -2,81 +2,40 @@ package pass
 
 import (
 	"context"
-	"fmt"
-	"github.com/gopasspw/gopass/pkg/store/root"
-	"log"
-	"os"
 	"sync"
 
-	"github.com/blang/semver"
-	"github.com/gopasspw/gopass/pkg/action"
-	_ "github.com/gopasspw/gopass/pkg/backend/storage"
-	"github.com/gopasspw/gopass/pkg/config"
-	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
-	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/gopasspw/gopass/pkg/gopass/api"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 	"github.com/pkg/errors"
 )
 
-type PassProvider struct {
-	store *root.Store
+type passProvider struct {
+	store *api.Gopass
 	mutex *sync.Mutex
 }
 
-func Provider() terraform.ResourceProvider {
+func Provider() *schema.Provider {
 	return &schema.Provider{
-		Schema: map[string]*schema.Schema{
-			"store_dir": {
-				Type:        schema.TypeString,
-				Optional:    true,
-				DefaultFunc: schema.EnvDefaultFunc("PASSWORD_STORE_DIR", ""),
-				Description: "Password storage directory to use.",
-			},
-			"refresh_store": {
```

Comment:
Is the new behavior to never refresh, or to refresh always?

Reply:
Idea is to never refresh.

It was set so by original author, it came from this ticket: https://github.com/camptocamp/terraform-provider-pass/issues/42 , refresh is broken if local store has no remote.

Reply:
Well the new public API just doesn't allow refresh (anymore).

Reply:
Right, it seems the new API doesn't let you access Git at all (which is problematic imo).